### PR TITLE
Fix the "set" (number in the code) sort order for decks.

### LINKED ISSUE #65: "Sort by set" displaying cards out of order.
Using the sort by set function on the following decklist puts Quandary and Yagura out of order (Quandary is mistakenly shown first):

https://netrunnerdb.com/en/decklist/38388/bloom-nobly-ink-black-cherry-blossoms#

### PR DESCRIPTION
**Pull request by plural**
This needs to make a synthetic key from the cycle and the pack positions
in order to sort properly.

Fixes #65 

I'm sleepy now so i don't know why the Set option was the only one that needed a fix from sorting the rows coming back from displayDescription[colnum] so i'll take a closer look tomorrow.

### CODE REVIEW

**Comment by NoahTheDuke** [comment]
Forgive me for not understanding, but can you give an example of how the old pack values and your new synthetic key are different? How does your fix work?

**Comment by plural** [comment]
Hey there! Noahpology necessary.  /me hides

This changes from a sort by pack code, which is effectively random, to one by cycle and set, so chronologically by release order.  We could probably consider treating the cores a little differently so they were grouped, but that is probably best left to a different change.

In SQL form, the old version did this:
```
SELECT code, name FROM pack ORDER BY code LIMIT 25;
+-------+----------------------+
| code  | name                 |
+-------+----------------------+
| 23s   | 23 Seconds           |
| asis  | A Study in Static    |
| atr   | All That Remains     |
| baw   | Blood and Water      |
| bb    | Breaker Bay          |
| bf    | Business First       |
| bm    | Blood Money          |
| cac   | Creation and Control |
| cc    | Chrome City          |
| cd    | Crimson Dust         |
| ce    | Cyber Exodus         |
| core  | Core Set             |
| core2 | Revised Core Set     |
| cotc  | Council of the Crest |
| dad   | Data and Destiny     |
| dag   | Democracy and Dogma  |
| dc    | Daedalus Complex     |
| draft | Draft                |
| dt    | Double Time          |
| dtwn  | Down the White Nile  |
| eas   | Earth's Scion        |
| es    | Escalation           |
| fal   | Fear and Loathing    |
| fc    | First Contact        |
| fm    | Free Mars            |
+-------+----------------------+
```
which is a bit arbitrary, instead of this:
```
SELECT c.position, c.name, p.position, p.name FROM cycle c INNER JOIN pack p ON c.id = p.cycle_id LIMIT 25;
+----------+----------------------+----------+----------------------+
| position | name                 | position | name                 |
+----------+----------------------+----------+----------------------+
|        0 | Draft                |        1 | Draft                |
|        1 | Core Set             |        1 | Core Set             |
|        2 | Genesis              |        4 | A Study in Static    |
|        2 | Genesis              |        3 | Cyber Exodus         |
|        2 | Genesis              |        6 | Future Proof         |
|        2 | Genesis              |        5 | Humanity's Shadow    |
|        2 | Genesis              |        2 | Trace Amount         |
|        2 | Genesis              |        1 | What Lies Ahead      |
|        3 | Creation and Control |        1 | Creation and Control |
|        4 | Spin                 |        6 | Double Time          |
|        4 | Spin                 |        5 | Fear and Loathing    |
|        4 | Spin                 |        3 | Mala Tempora         |
|        4 | Spin                 |        1 | Opening Moves        |
|        4 | Spin                 |        2 | Second Thoughts      |
|        4 | Spin                 |        4 | True Colors          |
|        5 | Honor and Profit     |        1 | Honor and Profit     |
|        6 | Lunar                |        5 | All That Remains     |
|        6 | Lunar                |        3 | First Contact        |
|        6 | Lunar                |        6 | The Source           |
|        6 | Lunar                |        2 | The Spaces Between   |
|        6 | Lunar                |        4 | Up and Over          |
|        6 | Lunar                |        1 | Upstalk              |
|        7 | Order and Chaos      |        1 | Order and Chaos      |
|        8 | SanSan               |        2 | Breaker Bay          |
|        8 | SanSan               |        3 | Chrome City          |
+----------+----------------------+----------+----------------------+
```

In screenshot form here is the before:
sort-by-set-before

And the after:
sort-by-set-after

File: web/js/nrdb.js [hunk 1]
```diff
@@ -300,6 +306,9 @@ function update_deck(options) {
     var cols_size = 12 / displayDescription.length;
     for (var colnum = 0; colnum < displayDescription.length; colnum++) {
         var rows = displayDescription[colnum];
+        // Don't rely on the rows being put into displayDescription in order.
+        rows.sort((a,b) => (a.id > b.id) ? 1 : ((b.id > a.id) ? -1 : 0));
```
Comment (NoahTheDuke) [inline web/js/nrdb.js]: This is just a readability thing, but would you mind opening this up to not be a nested ternary? I initially didn't see that there were two levels.

Comment (plural) [reply web/js/nrdb.js]: Good call.  I made this a more straightforward sort function and ditched the ternaries.

**Comment by NoahTheDuke** [comment]
Nice! Merged, and in production!

**Comment by NoahTheDuke** [comment]
From prod: 
image

**Pull request merged**